spring-integration-core/src/main/java/org/springframework/integration/store/MessageStore.java

```diff
@@ -39,6 +40,15 @@
 	Message<?> getMessage(UUID id);
 
 	/**
+	 * Return a {@link MessageMetadata} for the {@link Message} by provided {@code id}.
+	 * @param id The message identifier.
+	 * @return The MessageMetadata with the given id, or <i>null</i>
+	 * if no Message with that id exists in the MessageStore.
+	 * @since 5.0
+	 */
+	MessageMetadata getMessageMetadata(UUID id);
```

Comment:
Perhaps...
```
/**
 * Return a {@link MessageMetadata} for the {@link Message} by provided {@code id}.
 * @param id The message identifier.
 * @return The MessageMetadata with the given id, or <i>null</i>
 * if no Message with that id exists in the MessageStore
 * or the message has no metadata (legacy message from an earlier version).
 * @since 5.0
 */
```

If you agree, I can fix on merge.